src/test/java/unit/in/ac/bits/protocolanalyzer/persistence/repository/SaveRepositoryTest.java

```diff
@@ -1,59 +1,154 @@
+import org.mockito.runners.MockitoJUnitRunner;
 import org.springframework.data.elasticsearch.core.ElasticsearchTemplate;
 import org.springframework.data.elasticsearch.core.query.IndexQuery;
-import org.springframework.test.context.ContextConfiguration;
-import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
-import org.springframework.test.context.support.AnnotationConfigContextLoader;
 
 import com.google.common.eventbus.EventBus;
 
+import in.ac.bits.protocolanalyzer.analyzer.event.EndAnalysisEvent;
 import in.ac.bits.protocolanalyzer.persistence.repository.SaveRepository;
-import unit.config.in.ac.bits.protocolanalyzer.persistence.repository.SaveRepositoryTestConfig;
 
-@RunWith(SpringJUnit4ClassRunner.class)
-@ContextConfiguration(classes = SaveRepositoryTestConfig.class, loader = AnnotationConfigContextLoader.class)
+@RunWith(MockitoJUnitRunner.class)
 public class SaveRepositoryTest {
-	@Autowired
+	@Mock
+	public ElasticsearchTemplate template;
+	
+	@Spy
+	public ConcurrentLinkedQueue<ArrayList<IndexQuery>> buckets;
+	
+	@Mock
+	public Context ctx;
+	
+	@Mock
+	public Runtime runtime;
+	
+	@InjectMocks
 	public SaveRepository saveRepo;
+	
+	@Mock
+	public Context ctxString;
 
 	@Mock
 	public EventBus bus;
-
+	
 	@Mock
-	public ElasticsearchTemplate template;
-
+	public ArrayList<IndexQuery> listIndexQuery;
+	
+	@Mock
+	public EndAnalysisEvent event;
+	
+	
+	public long bytes = 123312312122L;
+	
 	@Before
-	public void setup() throws Exception {
+	public void setup() {
 		MockitoAnnotations.initMocks(this);
+	}
+	
+	
+	@Test
+	public void configureTest() {
+		doNothing().when(bus).register(saveRepo);
+		try {
+			when(ctx.lookup("java:comp/env")).thenReturn(ctxString);
+			when(ctxString.lookup("lowWaterMark")).thenReturn("2");
+			when(ctxString.lookup("analysisOnly")).thenReturn("true");
+		} catch (NamingException e) {
+			e.printStackTrace();
+		}
+		
 		saveRepo.configure(bus);
+		verify(bus).register(saveRepo);
+		
+		try {
+			verify(ctx,times(2)).lookup("java:comp/env");
+			verify(ctxString,times(1)).lookup("lowWaterMark");
+			
+		} catch (NamingException e) {
+			e.printStackTrace();
+		}
+		
+		assertThat(saveRepo.getLowWaterMark(), equalTo(2));
+		assertThat(saveRepo.isAnalysisOnly(), equalTo(true));
+		assertNotNull(saveRepo);
 	}
-
+	
 	@Test
-	public void wiringTest() {
-		assertThat(saveRepo,is(notNullValue()));
+	public void bytesToMegabytesTest() {
+		assertThat(SaveRepository.bytesToMegabytes(bytes), equalTo(bytes/(1024L*1024L)));
 	}
-
+	
+	
+	@Test
+	public void runTest() {
+		saveRepo.setBucket(listIndexQuery);
+		saveRepo.setAnalysisOnly(true);
+		saveRepo.analysisRunning = false;
+		saveRepo.setLowWaterMark(2);
+		
+		when(runtime.totalMemory()).thenReturn(10L);
+		when(runtime.freeMemory()).thenReturn(1L);
+		
+		ExecutorService executorService = Executors.newSingleThreadExecutor();
+		Future<?> future = executorService.submit(saveRepo);
+		try {
+			if(future.get()==null) {

```

Comment:
The meaning of this condition is not obvious to me. Please explain using a comment.